scalyr_agent/log_line_debug.py

```diff
@@ -0,0 +1,258 @@
+"""
+This module is used to print additional debug statements for the log-processing abstractions when they read log lines
+for a particular log file.
+"""
+
+from __future__ import unicode_literals
+
+from __future__ import absolute_import
+import io
+
+from scalyr_agent import scalyr_logging, line_matcher, log_processing, scalyr_client
+from scalyr_agent import configuration
+from scalyr_agent import compat
+
+log = scalyr_logging.getLogger(__name__)
+
+# The prefix for the pod which lines have to be tracked.
+POD_NAME = "lwbi-exporter-"
+
+# the kubernetes monitor will write the id of the needed contained in this variable.
+CONTAINER_ID = None
+
+
+def create_message(method):
+    """
+
+    Create base message with the base fields.
+
+    """
+    data_str = "METHOD: {0}".format(method)
+
+    return data_str
+
+
+def log_message(method, logger, message):
+    """
+    Log custom message.
+    """
+    logger.info(", ".join([create_message(method), message]))
+
+
+def print_line_data(method, logger, line, print_line=False, additional_data=None):
+    """
+    Log info about the log line.
+    """
+    data_str = create_message(method)
+
+    count = len(line)
+
+    data_str = ", ".join([data_str, "COUNT: {0}".format(count)])
+
+    if print_line:
+        data_str = ", ".join([data_str, "LINE: {0}".format(line)])
+
+    if additional_data is None:
+        additional_data = {}
+
+    for k, v in additional_data.items():
+        data_str = ", ".join([data_str, "{0}: {1}".format(k.upper(), v)])
+
+    logger.info(data_str)
+
+
+class DebugConfiguration(configuration.Configuration):
+    """
+    The subclass of the configuration needed to specify additional debug options.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super(DebugConfiguration, self).__init__(*args, **kwargs)
+
+    def parse(self):
+        global POD_NAME
+        super(DebugConfiguration, self).parse()
+        self._Configuration__verify_or_set_optional_string(
+            self._Configuration__config,
+            "_debug_log_path",
+            POD_NAME,
+            "",
+            env_aware=True,
+        )
+
+        POD_NAME = self._debug_log_path
+
+    @property
+    def _debug_log_path(self):
+        return self._Configuration__get_config().get_string("_debug_log_path")
+
+
+class DebugMixin:
+    """
+    The debug purpose class which is used to add debug logic to log-processing abstractions.
+    """
+
+    def __init__(self):
+        # logger to write debug statements.
+        self._debug_logger = log
+
+        # only print debug statements if log-processing abstractions are processing this file.
+        self._debug_log_path = None
+
+        # global config.
+        self._config = None
+
+    def set_debug_attributes(self, log_file_path, config):
+        self._debug_log_path = log_file_path
+        self._config = config
+
+    @property
+    def is_not_debug_file(self):
```

Comment:
Why not write this as `is_debug_file`?

Reply:
It's just avoid using `not` on every property access 😅  The main purpose of this property is to verify if the current log file is not the log file that we want, so we can exit immediately.

https://github.com/scalyr/scalyr-agent-2/blob/1ac1ca901c5f5c1ce7da7a04239917ed47c24332/scalyr_agent/log_line_debug.py#L128-L132